.github/workflows/dependencies/updater.py

```diff
@@ -14,437 +19,559 @@
-    response = requests.get(url)
-
-    # If the request was successful
-    if response.status_code == 200:
-      # Parse the JSON response
-      data = response.json()
-
-      # If the base is behind the head, there is a newer version
-      has_updates = data["status"] != "identical"
-
-      if not has_updates:
-        return {
-          "has_updates": False,
-        }
-
-      return {
-        "has_updates": data["status"] != "identical",
-        "version": data["commits"][-1]["sha"],
-        "compare_url": data["permalink_url"],
-        "head_ref": data["commits"][-1]["sha"],
-        "head_url": data["commits"][-1]["html_url"]
-      }
-    else:
-      # If the request was not successful, raise an exception
-      raise Exception(f"GitHub API request failed with status code {response.status_code}: {response.json()}")
-
-  @staticmethod
-  def create_issue(title: str, body: str) -> None:
-    cmd = [
-      "gh",
-      "issue",
-      "create",
-      "-t", title,
-      "-b", body
-    ]
-    CommandRunner.run_or_fail(cmd, stage="CreateIssue")
-
-  @staticmethod
-  def create_pr(branch: str, title: str, body: str) -> None:
-    cmd = [
-      "gh",
-      "pr",
-      "create",
-      "-B", Git.default_branch,
-      "-H", branch,
-      "-t", title,
-      "-b", body
-    ]
-    CommandRunner.run_or_fail(cmd, stage="CreatePullRequest")
+    @staticmethod
+    def check_newer_tag(repo, current_tag) -> UpdateStatusFalse | UpdateStatusTrue:
+        # GET /repos/:owner/:repo/git/refs/tags
+        url = f"https://api.github.com/repos/{repo}/git/refs/tags"
+
+        # Send a GET request to the GitHub API
+        response = requests.get(url)
+
+        # If the request was successful
+        if response.status_code == 200:
+            # Parse the JSON response
+            data = response.json()
+
+            if len(data) == 0:
+                return {
+                    "has_updates": False,
+                }
+
+            latest_ref = None
+            latest_version: Optional[Version] = None
+            for ref in data:
+                # we the tag since GitHub returns it as plain git ref
+                tag_version = coerce(ref["ref"].replace("refs/tags/", ""))
+                if tag_version is None:
+                    # we skip every tag that is not semver-complaint
+                    continue
+                if latest_version is None or tag_version.compare(latest_version) > 0:
+                    # if we have a "greater" semver version, set it as latest
+                    latest_version = tag_version
+                    latest_ref = ref
+
+            # raise if no valid semver tag is found
+            if latest_ref is None:
+                raise ValueError(f"No tags following semver found in {repo}")
+
+            # we get the tag since GitHub returns it as plain git ref
+            latest_tag = latest_ref["ref"].replace("refs/tags/", "")
+
+            if latest_tag == current_tag:
```

Comment:
Needs checking if the found version is actually semver-greater than the current one we have (e.g. if the version tag we have is removed and the latest one is now one version less).